src/toxcore/friend_connection/packet/share_relays.rs

```diff
@@ -0,0 +1,68 @@
+/*! ShareRalays struct
+*/
+
+use toxcore::binary_io::*;
+use toxcore::dht::packed_node::*;
+
+/** ShareRelays is a struct that holds 3 TCP relays as a PackedNode format.
+
+This packet is used to share relay info. between two friends.
+
+Serialized form:
+
+Length    | Content
+--------- | ------
+`1`       | `0x11`
+variable  | 3 PackedNodes
+
+*/
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct ShareRelays {
+    relays: Vec<PackedNode>,
+}
+
+impl FromBytes for ShareRelays {
+    named!(from_bytes<ShareRelays>, do_parse!(
+        tag!("\x11") >>
+        relays: count!(PackedNode::from_tcp_bytes, 3) >>
+        (ShareRelays {
+            relays: relays,
+        })
+    ));
+}
+
+impl ToBytes for ShareRelays {
+    fn to_bytes<'a>(&self, buf: (&'a mut [u8], usize)) -> Result<(&'a mut [u8], usize), GenError> {
+        do_gen!(buf,
+            gen_be_u8!(0x11) >>
+            gen_cond!(self.relays.len() != 3, |buf| gen_error(buf, 0)) >>
```

Comment:
Does it mean that we have to share exactly 3 relays?

Reply:
It should have at most 3 relays.

Reply:
Then the condition should be `<= 3`, shouldn't it?

Reply:
> gen_cond!(bool, I -> Result<I,E>) => I -> Result<I,E> Conditional combinator

if condition is true then closure runs. So if len() > 3 then generate error.